env0/data_oidc_credentials.go

```diff
@@ -0,0 +1,55 @@
+package env0
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataOidcCredentials(credentialsType CloudType) *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataOidcCredentialRead(credentialsType),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Description:  fmt.Sprintf("the name of the %s oidc credentials", credentialsType),
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+			},
+			"id": {
+				Type:         schema.TypeString,
+				Description:  fmt.Sprintf("the id of the %s oidc credentials", credentialsType),
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+			},
+		},
+	}
+}
+
+func dataOidcCredentialRead(credentialsType CloudType) func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+		var credentials client.Credentials
+		var err error
+
+		id, ok := d.GetOk("id")
+		if ok {
+			credentials, err = getCredentialsById(id.(string), credentialsTypeToPrefixList[credentialsType], meta)
+		} else {
+			credentials, err = getCredentialsByName(d.Get("name").(string), credentialsTypeToPrefixList[credentialsType], meta)
```

Comment:
don't we want to check that `d.Get("name")` is ok?

Reply:
not required. This is enforced in the schema validation level ("exactlyOneOf").